kong/router.lua

```diff
@@ -1236,13 +1257,29 @@ function _M.new(routes)
 
     req_method = upper(req_method)
 
-    if req_host ~= "" then
-      -- strip port number if given because matching ignores ports
-      local idx = find(req_host, ":", 2, true)
-      if idx then
-        ctx.req_host = sub(req_host, 1, idx - 1)
+    -- req_host might have port or maybe not, host_no_port definitely doesn't
+    -- if there wasn't a port, req_port is assumed to be the default port
+    -- according the protocol scheme
+    local host_with_port = req_host
+    local host_no_port, req_port = utils.split_port(host_with_port)
+    if not req_port then
```

Comment:
Consider putting the port defaults and the calculation of the new `host_with_port` inside `utils.split_port` (at that point it would probably make sense to move it out of `utils` and inside the router itself as a local). That way we can use memoization for that part as well.

(I would never recommend this avenue if this wasn't a hot path)

Reply:
done